### ISSUE TITLE
Add option to retrieve metadata from www.wlnupdates.com or www.novelupdates.com

### ISSUE THREAD

**Issue by Emasoft** [MakeYourLifeEasier/novel-ebook.com#100]
Can you add the option to retrieve the novel metadata (title, original title, alternative titles, author name, original author name, publisher, translator, first published date, original language, summary, genres, etc. ) from www.wlnupdates.com or www.novelupdates.com and save it in the epub?

**Comment by MichaelGuardian** [comment]
While it's an interesting idea, I'm currently not able to put much development time into this project. Besides I inquired about the possibility of retrieving metadata from novelupdates.com and they were opposed from scraping information from their side. Also I'm pretty sure that the epub spec doesn't account for this kind of metadata, so retrieving it wouldn't be convenient in the first place.

**Comment by Emasoft** [comment]
This is not true.
You can find many scrapers for novelupdates:
https://github.com/shaido987/novel-dataset-scraper
https://github.com/mHaisham/novelsave
https://github.com/ajdroid270/WebNovelBot-Modified
And more importantly, there are websites mirroring novelupdates that support an official API:
http://www.wlnupdates.com/
API: https://github.com/fake-name/wlnupdates/blob/master/app/templates/api-docs.md

**Comment by Emasoft** [comment]
Also ePub files have extensive metadata, and the best ePub readers (i.e. KyBook3 on iOS) can show metadata and sort ePubs via those for easier navigation. 
Here is a real world example of the epub output:

```xml
<dc:identifier id="book-id" opf:scheme="ISBN">1234567890X</dc:identifier> 
    <dc:title id="english">Battle Through the Heavens</dc:title>
    <meta refines="#english" property="title-type">english title</meta>
    <dc:title id="original">斗破苍穹</dc:title>
    <meta refines="#original" property="title-type">original title</meta>
    <dc:title id="alternative">Fights Break Sphere</dc:title>
    <meta refines="#alternative" property="title-type">alternative title</meta>
     <dc:language id="text-language">en</dc:language>
    <meta refines="#text-language" property="identifier-type" scheme="onix:codelist22">01</meta>
    <dc:language id="original-language">cn</dc:language>
    <meta refines="#original-language" property="identifier-type" scheme="onix:codelist22">02</meta>
    <dc:creator opf:role="aut" >Heavenly Silkworm Potato</dc:creator>
    <dc:creator opf:role="aut" >Tian Can Tu Dou</dc:creator>
    <dc:creator opf:role="aut" >天蚕土豆</dc:creator>
    <dc:creator opf:role="trl" >GravityTales</dc:creator>
    <dc:contributor opf:role="ill" >Hongbin Zhou</dc:contributor>
    <dc:publisher>Qidian</dc:publisher>
    <dc:subject>Action</dc:subject>
    <dc:subject>Adventure</dc:subject>
    <dc:subject>Fantasy</dc:subject>
    <dc:subject>Harem</dc:subject>
    <dc:subject>Martial Arts</dc:subject>
    <dc:subject>Xuanhuan</dc:subject>
    <dc:date opf:event="publication">2018-01-01T00:00:00Z</dc:date>
    <dc:source>urn:isbn:1234567890X</dc:source>
    <dc:description>"In a land where no magic is present. A land where the strong make the rules and the weak have to obey. A land filled with alluring treasures and beauty, yet also filled with unforeseen danger. Three years ago, Xiao Yan, who had shown talents none had seen in decades, suddenly lost everything. His powers, his reputation, and his promise to his mother. What sorcery has caused him to lose all of his powers? And why has his fiancee suddenly shown up? </dc:description>
 <link href="https://www.novelupdates.com/series/battle-through-the-heavens/" />
  </metadata> 
```

All the metadata in the example above is taken from Novelupdates.
An extensive API is provided on both websites. You can check wlnupdates.com API here:

https://github.com/fake-name/wlnupdates/blob/master/app/templates/api-docs.md

**Comment by MichaelGuardian** [comment]
Thanks for elaborating!
I'm aware of of those scrapers but decided against using them since I talked with the admin of novelupdates and he asked me not to use them.

That being said I was unaware of how far wlnupdates has come since the last time I checked in on it and did never see an e reader that used metadata as extensively as KyBook.

I put this feature on my planning board and will take a closer look at the api and how to integrate it with the current server structure once I have time to spend on this project.
